src/utils/Cypher.ts

```diff
@@ -26,19 +35,193 @@ export default class Cypher {
     return this.configuration;
   }
 
-  public static encrypt(data: string): string {
+  public static async migrateCryptoKey(tenantID: string): Promise<void> {
+    // Crypto Key from config file
+    const configCryptoKey: string = Configuration.getCryptoConfig().key;
+    // Crypto Key Settings from config file
+    const configCryptoKeySettings: KeyCryptoSetting = {
+      blockCypher: Configuration.getCryptoConfig().blockCypher,
+      keySize: Configuration.getCryptoConfig().keySize,
+      operationMode: Configuration.getCryptoConfig().operationMode
+    };
+
+    // Crypto Key Setting from db
+    const keySettings = await SettingStorage.getCryptoSettings(tenantID);
+
+    // If no Crypto Key Setting exist, initialize them with Crypto Key from config file
+    if (!keySettings) {
+      // Create New Crypto Key in Tenant Settings
+      const keySettingToSave = {
+        identifier: TenantComponents.CRYPTO,
+        type: CryptoSettingsType.CRYPTO,
+        crypto: {
+          key: configCryptoKey,
+          keySetting: configCryptoKeySettings,
+          migrationDone: true
+        }
+      } as KeySettings;
+      await SettingStorage.saveCryptoSettings(tenantID, keySettingToSave);
+    }
+  }
+
+  public static getAlgorithm(key: string): string {
+    let algorithm: string;
+    if (key === this.cryptoSetting?.formerKey) {
+      algorithm = `${this.cryptoSetting.formerKeySetting.blockCypher.toLowerCase()}-${this.cryptoSetting.formerKeySetting.keySize}-${this.cryptoSetting.formerKeySetting.operationMode.toLowerCase()}`;
+    } else if (key === this.cryptoSetting?.key) {
+      algorithm = `${this.cryptoSetting.keySetting.blockCypher.toLowerCase()}-${this.cryptoSetting.keySetting.keySize}-${this.cryptoSetting.keySetting.operationMode.toLowerCase()}`;
+    }
+    return algorithm;
+  }
+
+  public static getCryptoKeySync(): string {
+    return this.cryptoSetting?.migrationDone ? this.cryptoSetting?.key : this.cryptoSetting?.formerKey;
+  }
+
+  public static async getCryptoKey(tenantID: string): Promise<string> {
+    if (!this.cryptoSetting) {
+      this.keySetting = await SettingStorage.getCryptoSettings(tenantID);
+      this.cryptoSetting = this.keySetting.crypto;
+    }
+    return this.cryptoSetting?.migrationDone ? this.cryptoSetting?.key : this.cryptoSetting?.formerKey;
```

Comment:
This is wrong, during the migration you can have settings migrated and other not.
To review the full algo with me and to simplify.